py/frequenz/client/electricity_trading/_types.py

```diff
@@ -886,15 +886,17 @@ def from_pb(cls, order: electricity_trading_pb2.Order) -> Self:
             side=MarketSide.from_pb(order.side),
             price=Price.from_pb(order.price),
             quantity=Energy.from_pb(order.quantity),
-            stop_price=Price.from_pb(order.stop_price) if order.stop_price else None,
+            stop_price=Price.from_pb(order.stop_price)
+            if order.HasField("stop_price")
+            else None,
             peak_price_delta=Price.from_pb(order.peak_price_delta)
-            if order.peak_price_delta
+            if order.HasField("peak_price_delta")
             else None,
             display_quantity=Energy.from_pb(order.display_quantity)
-            if order.display_quantity
+            if order.HasField("display_quantity")
             else None,
             execution_option=OrderExecutionOption.from_pb(order.execution_option)
-            if order.execution_option
+            if order.HasField("execution_option")
             else None,
             valid_until=order.valid_until.ToDatetime() if order.valid_until else None,
             payload=json_format.MessageToDict(order.payload) if order.payload else None,
```

Comment:
I presume the check `order.HasField({field})` is needed for calling `{Class}.from_pb(order.{field})`, otherwise a similar check should have been applied here. You might want to amend/clarify it in the commit message if that's the case

Reply:
I've updated the commit message